cmd/server/main.go

```diff
@@ -62,6 +65,210 @@ type BindingSecret struct {
+			err := handleDeleteServiceBinding(clientConfig.ServiceBindingId)
+			if err != nil {
+				log.Printf("Error deleting service binding instance with id %s\n%s", clientConfig.ServiceBindingId, err.Error())
+			}
+		}
+	}
+}
+
+// getUPSSecrets() returns a list of the secrets that contain the UPS client configs
+func getUPSSecrets() ([]v1.Secret, error) {
+	selector := fmt.Sprintf("serviceName=ups,pushApplicationId=%s", pushClient.config.ApplicationId)
+	filter := metav1.ListOptions{LabelSelector: selector}
+	secretsList, error := k8client.CoreV1().Secrets(os.Getenv(NamespaceKey)).List(filter)
+	return secretsList.Items, error
+}
+
+// getUPSVariantServiceBindingMappings() takes the list of secrets and returns a list of VariantServiceBindingMappings
+func getUPSVariantServiceBindingMappings(secrets []v1.Secret) []VariantServiceBindingMapping {
+
+	results := []VariantServiceBindingMapping{}
+
+	buildAndAppendResult := func(results []VariantServiceBindingMapping, variantId string, serviceBindingId string, secret v1.Secret) []VariantServiceBindingMapping {
+		if variantServiceBindingMapping, err := GetClientConfigRepresentation(variantId, serviceBindingId); err != nil {
+			log.Printf("invalid android UPS client config found in secret %s reason: %s", secret.Name, err.Error())
+			return results
+		} else {
+			return append(results, variantServiceBindingMapping)
+		}
+	}
+
+	for _, secret := range secrets {
+
+		// Retrieve the current config as an object
+		clientConfig := UPSClientConfig{}
+		json.Unmarshal(secret.Data["config"], &clientConfig)
+
+		if clientConfig.Android != nil {
+			androidConfig := *clientConfig.Android
+			variantId := androidConfig["variantId"]
+			serviceBindingId := secret.ObjectMeta.Annotations["binding/android"]
+			results = buildAndAppendResult(results, variantId, serviceBindingId, secret)
+		}
+
+		if clientConfig.IOS != nil {
+			iOSConfig := *clientConfig.IOS
+			variantId := iOSConfig["variantId"]
+			serviceBindingId := secret.ObjectMeta.Annotations["binding/ios"]
+			results = buildAndAppendResult(results, variantId, serviceBindingId, secret)
+		}
+	}
+	return results
+}
+
+func handleDeleteServiceBinding(servicebindingId string) error {
+	serviceBindingName, err := getServiceBindingNameByID(servicebindingId)
+	if err != nil {
+		return err
+	}
+	err = deleteServiceBinding(serviceBindingName)
+	return err
+}
+
+// Find a service binding by its ExternalID
+func getServiceBindingNameByID(bindingId string) (string, error) {
+	// Get a list of all service bindings in the namespace and find the one with a matching ExternalID
+	// This is not very efficient and could be improved with a jsonpath query but it looks like client-go
+	// does not support jsonpath or at least I could not find any examples.
+	bindings, err := scclient.ServicecatalogV1beta1().ServiceBindings(os.Getenv(NamespaceKey)).List(metav1.ListOptions{})
+	if err != nil {
+		return "", errors.New("Error listing service bindings")
```

Comment:
I wrote this, but maybe we should return the existing error here instead of creating a new one, hiding the real error message.